### PR TITLE
Alerting: Code-gen parsing of URL parameters and fix related bugs

### PR DESCRIPTION
**Pull request by alexweav**
**What this PR does / why we need it**:

In the past, there's been several PR comments about cases where we forget to read URL params from the context, mis-spell the key, etc. Discussion on how to keep the code and swagger doc in sync.

**Which issue(s) this PR fixes**:

This PR makes the generated code parse URL parameters in addition to request bodies. The params are passed in as arguments to route handlers, just like the body. Parameters are now guaranteed to be consistent with their definition and the compiler will catch any problems with parameters, with no developer intervention.

TLDR: The generated code now looks like:
```
--generated_base_api_provisioning.go

func (f *ForkedProvisioningApi) RouteGetMuteTiming(ctx *models.ReqContext) response.Response {
	nameParam := web.Params(ctx.Req)[":name"]
	return f.forkRouteGetMuteTiming(ctx, nameParam)
}
```

This PR also fixes a bug having to do with ignored parameters in a route in the Provisioning API. This catch demonstrates why this change is effective - this class of bug is prevented from happening in the future.

### CODE REVIEW

**Review by yuri-tceretian — APPROVED** [review]
LGTM! Nice change!

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM
